File: javascript/dfsPostOrder.js
class Node {
  constructor(value) {
    this.value = value;
    this.children = [];
  }
}

function dfs(node,values=[]) {
  node.children.forEach((child) => {
    values = values.concat(dfs(child));
  });
  return values.concat(node.value);
}


function buildTree() {
  const root = new Node(prompt("Enter the root node value:"));
  const queue = [root];

  while (queue.length > 0) {
    const node = queue.shift();
    const numChildren = +prompt(`Enter the number of children for node ${node.value} or (0: No children):`);

    for (let i = 1; i <= numChildren; i++) {
      const childNode = new Node(prompt(`Enter the value for child ${i} of node ${node.value}:`));
      node.children.push(childNode);
      queue.push(childNode);
    }
  }
  return root;
}

const root = buildTree();
console.log(`The result of DFS Search for the given tree is: ${dfs(root)}`);

File: python/dfsPostOrder.py
class Node:
    def __init__(self, value):
        self.value = value
        self.children = []

def post_order_traversal(node):
    if node is None:
        return

    for child in node.children:
        post_order_traversal(child)

    print(node.value, end=" ")  # Print the value of the node after visiting its children

def build_tree():
    root_value = input("Enter the root node value: ")
    root = Node(root_value)
    queue = [root]

    while queue:
        node = queue.pop(0)

        num_children = int(input(f"Enter the number of children for node {node.value} (or 0 if it has no children): "))

        for i in range(num_children):
            child_value = input(f"Enter the value for child {i+1} of node {node.value}: ")
            child_node = Node(child_value)
            node.children.append(child_node)
            queue.append(child_node)

    return root

root = build_tree()
post_order_traversal(root)
